Tests/ApolloTests/PromiseTests.swift

```diff
@@ -249,4 +249,34 @@ class PromiseTests: XCTestCase {
     
     waitForExpectations(timeout: 1)
   }
+
+  func testWhenAllRejectsWhenAnyOfThePromisesRejectsAsync_doesNotCreateMemoryLeak() throws {
+    var executeReject: ((Error) -> Void)?
+
+    var promises: [Promise<String>] = [Promise(fulfilled: "foo"),
+                                       Promise<String> { _, reject in executeReject = reject },
+                                       Promise(fulfilled: "bar")]
+    weak var weakPromise0 = promises[0]
+    weak var weakPromise1 = promises[1]
+    weak var weakPromise2 = promises[2]
+
+    let expectation = self.expectation(description: "whenAll catch handler invoked")
+
+    whenAll(promises).catch { error in
+      XCTAssert(error is TestError)
+
+      expectation.fulfill()
+    }
+
+    promises = []
+    executeReject?(TestError())
+    executeReject = nil
+
+    waitForExpectations(timeout: 1)
+
+    XCTAssertNil(weakPromise0)
+    XCTAssertNil(weakPromise1)
```

Comment:
So before this fix, this particular `assert` would fail, correct?

Reply:
Actually, all three of the asserts were failing. The notify block references the entire array of promises, so all of them are retained.

Reply:
oof, ouch. well this is definitely an improvement.